ffi/capi/bindings/js/ICU4XMeasureUnitParser.d.ts

```diff
@@ -12,10 +10,9 @@ export class ICU4XMeasureUnitParser {
 
   /**
 
-   * Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding {@link ICU4XMeasureUnit `ICU4XMeasureUnit`}. Returns an error if the unit identifier is not valid.
+   * Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding {@link ICU4XMeasureUnit `ICU4XMeasureUnit`}, if the identifier is valid.
 
    * See the {@link https://docs.rs/icu/latest/icu/experimental/units/measureunit/struct.MeasureUnitParser.html#method.parse Rust documentation for `parse`} for more information.
-   * @throws {@link FFIError}<{@link ICU4XError}>
    */
-  parse(unit_id: string): ICU4XMeasureUnit | never;
+  parse(unit_id: string): ICU4XMeasureUnit | undefined;
```

Comment:
Issue: I think our parse functions elsewhere (often constructors) return errors. Why return option here?

Reply:
because the error is a ZST 🤷

Reply:
(the error will be a ZST, currently it's a single enum variant, which doesn't add any information)

Reply:
What I mean is, when parse fails, I think it should stay an exception ?

Reply:
Exceptions are more difficult to handle in all languages than nullable values (i.e. in C++ you have `diplomat::result`, a non-standard type, instead of `std::optional`, in Dart you have exceptions declared in docs instead of a nullability in the type system, same in Kotlin), so I'd rather do nullable if the error type doesn't provide any additional information. Returning null on failure is not uncommon.

Reply:
I don't think a ZST is a good reason to change an exception to a falsy return value, but it seems okay in this particular case, since it is a method.

Reply:
We currently don't support ZST in HIR, I will add this for fine-grained error types and then we can change this back.